ci: post-release label step fails when a commit message references an issue by #NNNN

## Context

The `Post-Release / Label PRs with version` job failed after the 8.9 release:
https://github.com/camunda/camunda-platform-helm/actions/runs/24834134425/job/72689928743

```
GraphQL: Could not resolve to a PullRequest with the number of 5874.
[PR 5874] Labeling the PR with chart version label version:8.9-14.0.0 -
gh: Could not resolve to a PullRequest with the number of 5874.
make: *** [Makefile:334: release.set-prs-version-label] Error 1
```

`#5874` is actually an issue (#5874), not a PR. The reference comes from commit `e3f80d059 fix: #5874 remove prerelease annotation (#5875)`.

## Problem

`scripts/set-prs-version-label.sh:38` extracts every `#NNNN` token from commit messages:

```bash
jq '.[].commits[].message' | (grep -Po '(?<=#)\d+' || true)
```

Each extracted number is then passed to `gh pr edit`. When a commit body references an issue (`fix: #5874`), the script tries to edit it as a PR and the job fails the whole post-release labeling step.

## Solution

Before labeling, skip numbers that aren't PRs. Minimal fix in the loop at `scripts/set-prs-version-label.sh:95`:

```bash
gh pr view "${pr_nubmer}" --json number >/dev/null 2>&1 || {
    printf -- "- [#${pr_nubmer}] not a PR, skipping.\n"
    continue
}
```

This is more robust than narrowing the regex to the trailing `(#NNNN)` squash-merge pattern, since it also tolerates other commit-message styles (e.g. `Merge pull request #NNNN`) and any future stray issue refs.

## Acceptance criteria

- [ ] `scripts/set-prs-version-label.sh` skips `#NNNN` references that aren't PRs instead of failing
- [ ] The 8.9 post-release labeling is re-run (or patched up manually) so PR #5875 and issue #5874 carry the `version:8.9-14.0.0` label
- [ ] Subsequent post-release runs succeed without manual intervention

## References

- Failing run: https://github.com/camunda/camunda-platform-helm/actions/runs/24834134425/job/72689928743
- Offending commit: `e3f80d059`
- Script: `scripts/set-prs-version-label.sh`

--- a/.github/workflows/chart-release-public.yaml
+++ b/.github/workflows/chart-release-public.yaml
@@ -426,6 +426,7 @@ jobs:
 
       - name: Mark release-please PR as published
         id: mark-published
+        if: ${{ !cancelled() }}
         env:
           GH_TOKEN: ${{ steps.generate-github-token.outputs.token }}
           CHART_DIR_ID: "${{ needs.release.outputs.chart-dir-id }}"
@@ -453,6 +454,7 @@ jobs:
 
       - name: Approve and auto-merge release-please PR
         id: auto-merge
+        if: ${{ !cancelled() && steps.mark-published.outcome == 'success' }}
         continue-on-error: true
         env:
           GH_TOKEN: ${{ steps.generate-github-token.outputs.token }}
@@ -479,6 +481,7 @@ jobs:
           echo "enabled=true" >> $GITHUB_OUTPUT
 
       - name: Summary
+        if: ${{ !cancelled() }}
         env:
           GH_TOKEN: ${{ steps.generate-github-token.outputs.token }}
         run: |

--- a/scripts/set-prs-version-label.sh
+++ b/scripts/set-prs-version-label.sh
@@ -42,8 +42,8 @@ get_prs_per_chart_dir () {
 # Note: GH CLI doesn't support this so we use the API call directly.
 # https://github.com/cli/cli/discussions/7097#discussioncomment-5229031
 get_issues_per_pr () {
-    pr_nubmer="${1}"
-    gh api graphql -F owner="${REPO_OWNER}" -F repo="${REPO_NAME}" -F pr="${pr_nubmer}" -f query='
+    pr_number="${1}"
+    gh api graphql -F owner="${REPO_OWNER}" -F repo="${REPO_NAME}" -F pr="${pr_number}" -f query='
         query ($owner: String!, $repo: String!, $pr: Int!) {
             repository(owner: $owner, name: $repo) {
                 pullRequest(number: $pr) {
@@ -94,34 +94,34 @@ echo "${chart_dirs}" | while read -r chart_dir; do
     # Update GH PRs and Issues with the chart version.
     # Labeling is best-effort: a single failure (non-PR reference, transient API
     # error, etc.) must not abort the post-release workflow.
-    get_prs_per_chart_dir "${chart_dir}" | while read pr_nubmer; do
+    get_prs_per_chart_dir "${chart_dir}" | while read -r pr_number; do
         # Commit messages may reference issues by #NNNN too; skip those.
         # The REST pulls endpoint 404s for issue numbers (gh pr view would
         # silently fall back to issue view, so it can't be used here).
-        if ! gh api "repos/${REPO_OWNER}/${REPO_NAME}/pulls/${pr_nubmer}" --silent >/dev/null 2>&1; then
-            printf -- "- [#${pr_nubmer}] not a PR, skipping.\n"
+        if ! gh api "repos/${REPO_OWNER}/${REPO_NAME}/pulls/${pr_number}" --silent >/dev/null 2>&1; then
+            printf -- "- [#${pr_number}] not a PR, skipping.\n"
             continue
         fi
         # Label PR.
-        printf "[PR ${pr_nubmer}] Labeling the PR with chart version label ${chart_version_label}"
-        if ! pr_edit_out="$(gh pr edit "${pr_nubmer}" --add-label "${app_version_label},${chart_version_label}" 2>&1)"; then
-            printf -- "\n- [PR ${pr_nubmer}] labeling failed, skipping: %s\n" "${pr_edit_out}"
+        printf "[PR ${pr_number}] Labeling the PR with chart version label ${chart_version_label}"
+        if ! pr_edit_out="$(gh pr edit "${pr_number}" --add-label "${app_version_label},${chart_version_label}" 2>&1)"; then
+            printf -- "\n- [PR ${pr_number}] labeling failed, skipping: %s\n" "${pr_edit_out}"
             continue
         fi
         printf -- " - %s\n" "${pr_edit_out}"
         # Label the PR's corresponding issue.
-        if ! issue_nubmers="$(get_issues_per_pr ${pr_nubmer})"; then
-            printf -- "- [PR ${pr_nubmer}] could not fetch linked issues, skipping.\n"
+        if ! issue_numbers="$(get_issues_per_pr "${pr_number}")"; then
+            printf -- "- [PR ${pr_number}] could not fetch linked issues, skipping.\n"
             continue
         fi
-        test -z "${issue_nubmers}" && {
-            printf -- "- PR no. ${pr_nubmer} has no corresponding issues.\n"
+        test -z "${issue_numbers}" && {
+            printf -- "- PR no. ${pr_number} has no corresponding issues.\n"
             continue
         }
-        echo -e "${issue_nubmers}" | while read issue_nubmer; do
-            printf -- "- [Issue ${issue_nubmer}] Labeling the issue with chart version label ${chart_version_label}"
-            if ! issue_edit_out="$(gh issue edit "${issue_nubmer}" --add-label "${app_version_label},${chart_version_label}" 2>&1)"; then
-                printf -- "\n- [Issue ${issue_nubmer}] labeling failed, skipping: %s\n" "${issue_edit_out}"
+        echo -e "${issue_numbers}" | while read -r issue_number; do
+            printf -- "- [Issue ${issue_number}] Labeling the issue with chart version label ${chart_version_label}"
+            if ! issue_edit_out="$(gh issue edit "${issue_number}" --add-label "${app_version_label},${chart_version_label}" 2>&1)"; then
+                printf -- "\n- [Issue ${issue_number}] labeling failed, skipping: %s\n" "${issue_edit_out}"
                 continue
             fi
             printf -- " - %s\n" "${issue_edit_out}"

--- a/scripts/set-prs-version-label.sh
+++ b/scripts/set-prs-version-label.sh
@@ -92,19 +92,39 @@ echo "${chart_dirs}" | while read -r chart_dir; do
             --description "Issues and PRs related to chart version ${chart_version}"
 
     # Update GH PRs and Issues with the chart version.
+    # Labeling is best-effort: a single failure (non-PR reference, transient API
+    # error, etc.) must not abort the post-release workflow.
     get_prs_per_chart_dir "${chart_dir}" | while read pr_nubmer; do
+        # Commit messages may reference issues by #NNNN too; skip those.
+        # The REST pulls endpoint 404s for issue numbers (gh pr view would
+        # silently fall back to issue view, so it can't be used here).
+        if ! gh api "repos/${REPO_OWNER}/${REPO_NAME}/pulls/${pr_nubmer}" --silent >/dev/null 2>&1; then
+            printf -- "- [#${pr_nubmer}] not a PR, skipping.\n"
+            continue
+        fi
         # Label PR.
         printf "[PR ${pr_nubmer}] Labeling the PR with chart version label ${chart_version_label}"
-        printf -- " - %s\n" $(gh pr edit "${pr_nubmer}" --add-label "${app_version_label},${chart_version_label}")
+        if ! pr_edit_out="$(gh pr edit "${pr_nubmer}" --add-label "${app_version_label},${chart_version_label}" 2>&1)"; then
+            printf -- "\n- [PR ${pr_nubmer}] labeling failed, skipping: %s\n" "${pr_edit_out}"
+            continue
+        fi
+        printf -- " - %s\n" "${pr_edit_out}"
         # Label the PR's corresponding issue.
-        issue_nubmers="$(get_issues_per_pr ${pr_nubmer})"
+        if ! issue_nubmers="$(get_issues_per_pr ${pr_nubmer})"; then
+            printf -- "- [PR ${pr_nubmer}] could not fetch linked issues, skipping.\n"
+            continue
+        fi
         test -z "${issue_nubmers}" && {
             printf -- "- PR no. ${pr_nubmer} has no corresponding issues.\n"
             continue
         }
         echo -e "${issue_nubmers}" | while read issue_nubmer; do
             printf -- "- [Issue ${issue_nubmer}] Labeling the issue with chart version label ${chart_version_label}"
-            printf -- " - %s\n" $(gh issue edit "${issue_nubmer}" --add-label "${app_version_label},${chart_version_label}")
+            if ! issue_edit_out="$(gh issue edit "${issue_nubmer}" --add-label "${app_version_label},${chart_version_label}" 2>&1)"; then
+                printf -- "\n- [Issue ${issue_nubmer}] labeling failed, skipping: %s\n" "${issue_edit_out}"
+                continue
+            fi
+            printf -- " - %s\n" "${issue_edit_out}"
         done
     done
 done

